### PR TITLE
[v4] Better utils and factory

### PR DESCRIPTION
**Pull request by dylanseago**
Currently PaymentsUtils are useful, but could be better. Two additional features are needed:
- [x] Get fee rate recommendation (for when the full payments config isn't available)
- [x] Provide an address standardization/normalization method (to replace the deprecated address standardization service we currently use)

And PaymentsFactory would be more useful if it had the following because then applications would only have to declare coin-payments as an explicit dependency instead of all of the packages.
- [x] Method to instantiate PaymentsUtils and BalanceMonitor
- [x] Manage connections and reuse them across instantiated objects

Additionally, there's a bunch of repeated methods and code in certain places that should be moved around to reduce maintenance overhead and effort required to add new coins.

This PR is pretty massive, so a good starting point would be to look at the core interface changes, everything else is ultimately implementing that:
- [PaymentsUtils](https://github.com/go-faast/coin-payments/blob/v4/packages/payments-common/src/PaymentsUtils.ts)
- [PaymentsFactory](https://github.com/go-faast/coin-payments/blob/v4/packages/payments-common/src/PaymentsFactory.ts)

### CODE REVIEW

**Comment by dylanseago** [comment]
[Coin payments v4](https://trello.com/c/XijapWtg/8963-coin-payments-v4)

File: packages/coin-payments/src/constants.ts [hunk 1]
```diff
@@ -22,9 +18,6 @@ export const PAYMENTS_FACTORIES: {
   LTC: new LitecoinPaymentsFactory(),
   BCH: new BitcoinCashPaymentsFactory(),
   DOGE: new DogePaymentsFactory(),
-}
+} as const
```
Comment (dzdidi) [inline packages/coin-payments/src/constants.ts:21]: Why is it needed?

File: packages/ripple-payments/src/RipplePaymentsUtils.ts [hunk 2]
```diff
@@ -80,4 +93,20 @@ export class RipplePaymentsUtils extends RippleConnected implements PaymentsUtil
...
+
+  async getFeeRateRecommendation(level: AutoFeeLevels): Promise<FeeRate> {
+    let cushion: number | undefined
+    if (level === FeeLevel.Low) {
+      cushion = 1
+    } else if (level === FeeLevel.Medium) {
+      cushion = 1.2
+    } else if (level === FeeLevel.High) {
+      cushion = 1.5
+    }
```
Comment (dzdidi) [inline packages/ripple-payments/src/RipplePaymentsUtils.ts:105]: does it make sense to move this to consts?

File: packages/payments-common/src/PaymentsFactory.ts [hunk 3]
```diff
@@ -1,6 +1,85 @@
...
+          await connected.init()
+          return connected
+        } else {
+          // connection isnt cached yet, get it and add it to cache
+          const connected = instantiator(config)
+          await connected.init()
+          connections[url] = getConnection(connected)
+          return connected
+        }
+      }
```
Comment (dzdidi) [inline packages/payments-common/src/PaymentsFactory.ts:79]: Maybe something like this:
```typescript
 async initConnected<T extends U | P | B>(
    config: C,
    instantiator: (c: C) => T
  ): Promise<T> {
  return this.connectionManager ? reuseConnection(config) : createConnection(config)
}

private async createConnection(config) {
  const connected = instantiator(config)
  await connected.init()
  return connected
}

private async reuseConnection(config) {
  const { getConnection, getConnectionUrl, setConnection, connections } = this.connectionManager
  const url = getConnectionUrl(config)
  if (!url) {
    return createConnection(config)
  }
  const existingConnection = connections[url]
  if (existingConnection) {
    // connection is cached, pass it to instantiated object
    config = { ...config } // avoid mutating external objects
    setConnection(config, existingConnection)
    return createConnection(config)
  } else {
    // connection isnt cached yet, get it and add it to cache
    connected = createConnection(config)
    connections[url] = getConnection(connected)
    return connected
  }
}

```

**Review by dzdidi — APPROVED** [review]

**Pull request merged**